core/vm/evm.go

```diff
@@ -146,6 +149,19 @@ func NewEVM(blockCtx BlockContext, txCtx TxContext, statedb StateDB, chainConfig
 		chainRules:  chainConfig.Rules(blockCtx.BlockNumber, blockCtx.Random != nil, blockCtx.Time),
 	}
 	evm.interpreter = NewEVMInterpreter(evm)
+	evm.precompileManager = NewPrecompileManager(evm)
+
+	// register precompiles here
+
+	// e.g. register native minter to 0x0000000000000000000000000000000000001000
+	evm.precompileManager.Register(
+		common.HexToAddress("0x1000"),
+		nativeminter.NewNativeMinter(),
+	)
```

Comment:
do we want to have this functionality here? probably would be better to move to an example branch

Reply:
No, it def needs removed, at least registering one. Could probably just be moved to a precompile doc that explains a bit and shows examples